### PR TITLE
Add discord as a new provider

### PR DESCRIPTION
**Pull request by klarstil**
I added discord ([https://discordapp.com/developers/docs/topics/oauth2](https://discordapp.com/developers/docs/topics/oauth2)) as a new provider for bell. I'm working on a project which needs the ability to login with discord.

The tests are working & the code coverage is still 100%, here's a pastebin for the test results: [http://pastebin.com/CA7GnBdV](http://pastebin.com/CA7GnBdV).

The documentation of the provider was added to the `Providers.md` file as well.

Let me know if there's something missing.

### CODE REVIEW

File: lib/providers/discord.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        protocol: 'oauth2',
+        auth: 'https://discordapp.com/api/oauth2/authorize',
+        token: 'https://discordapp.com/api/oauth2/token',
+        scope: ['email', 'identify'], // see https://discordapp.com/developers/docs/topics/oauth2#scopes
+        profile: function (credentials, params, get, callback) {
+
+            get('https://discordapp.com/api/users/@me', null, (profile) => {
+
+                credentials.profile = {
+                    id: parseInt(profile.id),
```
Comment (ldesplat) [inline lib/providers/discord.js]: I am not sure I agree with the parseInt. If they return this id as a string we should honor that and not throw when they don't!

**Comment by ldesplat** [comment]
I apologize for the delay. If you agree with that one review comment, then I can merge this.

**Comment by klarstil** [comment]
Hey @ldesplat, no worries I was busy too. I updated the PR according to your CR & fixed a typo in the `Providers.md`.

**Comment by ldesplat** [comment]
Looks great. Thank you so much!

**Pull request merged**